cmd/uplink/cmd/cp.go

```diff
@@ -144,27 +144,50 @@ func download(ctx context.Context, src fpath.FPath, dst fpath.FPath, showProgres
 		return fmt.Errorf("destination must be local path: %s", dst)
 	}
 
-	metainfo, streams, err := cfg.Metainfo(ctx)
+	project, err := cfg.GetProject(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = project.Close()
+		if err != nil {
+			fmt.Printf("Error closing project uplink: %+v\n", err)
+		}
+	}()
+
+	var access libuplink.EncryptionAccess
+	copy(access.Key[:], []byte(cfg.Enc.Key))
+
+	bucket, err := project.OpenBucket(ctx, dst.Bucket(), &access, 0)
 	if err != nil {
 		return err
 	}
+	defer func() {
+		err := bucket.Close()
+		if err != nil {
+			fmt.Printf("Error closing bucket: %+v\n", err)
+		}
+	}()
 
-	readOnlyStream, err := metainfo.GetObjectStream(ctx, src.Bucket(), src.Path())
+	object, err := bucket.OpenObject(ctx, src.Path())
 	if err != nil {
 		return convertError(err, src)
 	}
 
-	download := stream.NewDownload(ctx, readOnlyStream, streams)
-	defer func() { err = errs.Combine(err, download.Close()) }()
+	rc, err := object.DownloadRange(ctx, 0, object.Meta.Size)
+	if err != nil {
+		return err
+	}
+	defer func() { err = errs.Combine(err, rc.Close()) }()
```

Comment:
On other places, the error returned by `Close()` is printed, but here it is combined with the existing error and returned. Perhaps, choose only one of the two approaches and make it consistent all over the code of Uplink CLI.